### DISCUSSION TITLE
Git Clone to not empty directory

### DISCUSSION THREAD

**Discussion by ryanmerolle** [gitpython-developers/GitPython#1496] [Q&A] [answered]
Hello,

Thanks for this package!

Instead of a traceback resulting, how would I notify the user for a non-empty directory when cloning a repo using the package?  Or would I just do better to check the directory first without gitpython?

Thanks!

**Comment by Byron** [comment, marked as answer]
> Or would I just do better to check the directory first without gitpython?

It should be possible to catch the exception as well if it's unique enough. Both will work, none is optimal, but that's something we'd have to live with.

**Reply by ryanmerolle** [reply]
I approached it by checking if the directory existed before I handed off to GitPython.

Thanks!
